docs/en/latest/practices/how-to-use-go-plugin-runner-in-apisix-ingress.md

```diff
@@ -0,0 +1,400 @@
+
+```bash
+/usr/local/apisix-go-plugin-runner/cmd/go-runner/go-runner
+```
+
+Please make a note of this address. We will use it in the rest of the configuration.
+
+#### Begin to build Docker Image
+
+Start building a Docker image based on the Dockerfile. The command is executed in the `/home/chever/api7/cloud_native/tasks/plugin-runner` directory. The command is as follows:
+
+```bash
+docker build -t apisix/forrunner:0.1 .
+```
+
+Command Explanation: Build an image with the name `apisix/forrunner` and mark it as version 0.1.
+
+#### Load the image to the cluster environment
+
+```bash
+kind load docker-image apisix/forrunner:0.1
+```
+
+Load the image into the kind cluster environment to pull the custom local image for installation during the helm installation.
+
+### Install APISIX Ingress
+
+#### Customize the helm chart
+
+This section focuses on modifying the `values.yaml` file in the official helm package so that it can install locally packaged images and run the `go-plugin-runner` executable properly.
+
+##### Fetch Official helm chart
+
+First, fetch the latest apisix helm chart package with the following command:
+
+```bash
+helm fetch apisix/apisix
+```
+
+The file tree is as follows:
+
+```bash
+chever@cloud-native-01:~/api7/cloud_native/tasks/plugin-runner$ tree -L 1
+.
+├── apisix-0.9.1.tgz
+└── apisix-go-plugin-runner
+
+1 directory, 1 file
+```
+
+##### Unzip
+
+Unzip the `apisix-0.9.1.tgz` file and prepare to rewrite the configuration. The unzip command is as follows.
+
+```bash
+tar zxvf apisix-0.9.1.tgz
+```
+
+The file tree is as follows:
+
+```bash
+chever@cloud-native-01:~/api7/cloud_native/tasks/plugin-runner$ tree -L 1
+.
+├── apisix
+├── apisix-0.9.1.tgz
+└── apisix-go-plugin-runner
+
+2 directories, 1 file
+```
+
+##### Change `values.yaml`
+
+Go to the `apisix` folder and modify the `values.yaml` file. The two changes are as follows:
+
+```yaml
+image:
+  repository: apisix/forrunner
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: 0.1
+```
+
+The first change sets the image for the helm installation to be a locally packaged image of your own.
+
+```yaml
+extPlugin:
+  enabled: true
+  cmd: ["/usr/local/apisix-go-plugin-runner/cmd/go-runner/go-runner", "run"]
+```
+
+The second change sets the position of go-runner in the container after running the container.
+
+##### Compress the modified helm chart
+
+Once configured, compress the `apisix` file. The compression command is as follows:
+
+```bash
+tar zcvf apisix.tgz apisix/
+```
+
+The compressed file is obtained, at which point the file tree is as follows:
+
+```bash
+chever@cloud-native-01:~/api7/cloud_native/tasks/plugin-runner$ tree -L 1
+.
+├── apisix
+├── apisix-0.9.1.tgz
+├── apisix-go-plugin-runner
+└── apisix.tgz
+
+2 directories, 2 files
+```
+
```

Comment:
unnecessary. Helm can use a directory

Reply:
This reason is not sufficient.
Even without compression, the result won't make any difference

Reply:
So the better way is to  Avoid using compression. 
 Got that, thank you.
I will fix it tomorrow
Have a good night.

Reply:
Solved~